### ISSUE TITLE
Unable to get reported results, loss 0 but accuracy ~11%

### ISSUE THREAD

**Issue by arsenious** [mveres01/Theano-LtRRE#2]
Hi
Thank you for the code

I was trying out your code but I'm having trouble replicating your results
Starting training...
Epoch 1 of 2 took 56.404s
  training loss:                0.042020
  validation loss:              0.024000
  validation accuracy:          96.88 %

.
.
.
.

What I'm getting:
Loading data and creating train/test splits... 
Starting training...
Epoch 1 of 20 took 271.095s
  training loss:		0.000000
  validation loss:		0.000000
  validation accuracy:		11.09 %

I've only changed epochs from 100 to 20 and nothing else. Please can you tell me the hyperparameters you used? Thanks

**Comment by arsenious** [comment]
More trace readings

Loading data and creating train/test splits... 
Starting training...
Epoch 1 of 20 took 271.095s
  training loss:		0.000000
  validation loss:		0.000000
  validation accuracy:		11.09 %
Epoch 2 of 20 took 273.249s
  training loss:		0.000000
  validation loss:		0.000000
  validation accuracy:		9.85 %
Epoch 3 of 20 took 289.966s
  training loss:		0.000000
  validation loss:		0.000000
  validation accuracy:		19.40 %
Epoch 4 of 20 took 287.455s
  training loss:		0.000000
  validation loss:		0.000000
  validation accuracy:		12.17 %
Epoch 5 of 20 took 345.286s
  training loss:		0.000000
  validation loss:		0.000000
  validation accuracy:		9.14 %
Epoch 6 of 20 took 364.354s
  training loss:		0.000000
  validation loss:		0.000000
  validation accuracy:		20.38 %
Epoch 7 of 20 took 350.133s
  training loss:		0.000000
  validation loss:		0.000000
  validation accuracy:		10.93 %
Epoch 8 of 20 took 357.000s
  training loss:		0.000000
  validation loss:		0.000000
  validation accuracy:		23.02 %
Epoch 9 of 20 took 352.193s
  training loss:		0.000000
  validation loss:		0.000000
  validation accuracy:		10.08 %
Epoch 10 of 20 took 338.582s
  training loss:		0.000000
  validation loss:		0.000000
  validation accuracy:		10.31 %
Epoch 11 of 20 took 350.096s
  training loss:		0.000000
  validation loss:		0.000000
  validation accuracy:		8.40 %
Epoch 12 of 20 took 336.969s
  training loss:		0.000000
  validation loss:		0.000000
  validation accuracy:		10.49 %
Epoch 13 of 20 took 335.868s
  training loss:		0.000000
  validation loss:		0.000000
  validation accuracy:		12.48 %
Epoch 14 of 20 took 356.898s
  training loss:		0.000000
  validation loss:		0.000000
  validation accuracy:		6.79 %
Epoch 15 of 20 took 322.949s
  training loss:		0.000000
  validation loss:		0.000000
  validation accuracy:		13.99 %
Epoch 16 of 20 took 351.228s
  training loss:		0.000000
  validation loss:		0.000000
  validation accuracy:		8.26 %

**Comment by mveres01** [comment]
Can you verify which versions of Python, numpy, theano, and lasagne you're using? e.g. 
```
python
Python 2.7.13 |Anaconda custom (64-bit)| (default, Dec 20 2016, 23:09:15)

>> import theano, lasagne, numpy
>> theano.__version__
'0.10.0dev1.dev-8a1af5b8ba718b1431dc05a8f079adf5f33e0694'
>> lasagne.__version__
'0.2.dev1'
>> numpy.__version__
'1.13.1'
```

I just downloaded from scratch and can confirm the default settings work as intended:

```
mveres@GPU11:/scratch/mveres/temp/Theano-LtRRE$ python mnist.py
Using cuDNN version 5110 on context None
Mapped name None to device cuda2: Tesla K40c (0000:04:00.0)
Loading data and creating train/test splits...
Starting training...
Epoch 1 of 20 took 31.462s
  training loss:                0.041663
  validation loss:              0.014919
  validation accuracy:          96.73 %
Epoch 2 of 20 took 31.400s
  training loss:                0.020967
  validation loss:              0.013716
  validation accuracy:          98.24 %
 ...
Epoch 6 of 20 took 31.349s
  training loss:                0.022138
  validation loss:              0.013227
  validation accuracy:          98.55 %
```

**Comment by arsenious** [comment]
Thank you for the quick reply
I'm using python 3.5 but lasagne changed theano to .8 from .9 
Any particular dependency on these specific versions? I'll try and fix them on python 3.5

**Comment by arsenious** [comment]
Lasagne .2 stable

**Comment by arsenious** [comment]
I'm able to replicate it now, thanks!
Steps I followed :  
1) Create a conda env 2.7 with anaconda packages
conda create -n theanoenv27 python=2.7 anaconda

2) Install bleeding edge version of theano and lasagne, as mentioned on lasagne's website
pip install --upgrade https://github.com/Theano/Theano/archive/master.zip
pip install --upgrade https://github.com/Lasagne/Lasagne/archive/master.zip

3) But this will get a theano error when importing theano
So reinstall using conda
pip uninstall -y theano
conda install -y theano

4) Clone this repo and run, should work

**Comment by mveres01** [comment]
Great! Glad things worked out. Closing this issue then.

**Comment by arsenious** [comment]
Thank you
